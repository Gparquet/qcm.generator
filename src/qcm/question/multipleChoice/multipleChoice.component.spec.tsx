import React from 'react';
import renderer from 'react-test-renderer';
import multipleChoiceFormatted from './mocks/multiple_choice_formatted';
import MultipleChoiceComponent from './multipleChoice.component';

describe('Choice Component', () => {
  const props = {
    id: 'multipleChoice',
    label: 'multiple choice',
    choices: multipleChoiceFormatted.choices,
    defaultChoice: '5f67c896-77b4-4b9b-bf19-fb2c8e274c68',
    onChange: () => {},
  };
  test('renders correctly', () => {
    const tree = renderer
      .create(<MultipleChoiceComponent {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
