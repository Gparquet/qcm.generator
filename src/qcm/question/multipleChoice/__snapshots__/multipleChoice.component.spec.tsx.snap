// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choice Component renders correctly 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <label
    className="MuiFormLabel-root-5"
  >
    multiple choice
  </label>
  <div
    className="MuiFormGroup-root-12"
  >
    <label
      className="MuiFormControlLabel-root-14"
    >
      <span
        className="MuiButtonBase-root-36 MuiIconButton-root-30 MuiPrivateSwitchBase-root-26 MuiCheckbox-root-20 MuiCheckbox-colorSecondary-25"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-35"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-39"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <input
            checked={false}
            className="MuiPrivateSwitchBase-input-29"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="5f67c896-77b4-4b9b-bf19-fb2c8e274c68"
          />
        </span>
        <span
          className="MuiTouchRipple-root-92"
        />
      </span>
      <span
        className="MuiTypography-root-48 MuiTypography-body1-57 MuiFormControlLabel-label-19"
      >
        Internal
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root-14"
    >
      <span
        className="MuiButtonBase-root-36 MuiIconButton-root-30 MuiPrivateSwitchBase-root-26 MuiCheckbox-root-20 MuiCheckbox-colorSecondary-25"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-35"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-39"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <input
            checked={false}
            className="MuiPrivateSwitchBase-input-29"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="8762b6dc-9fe3-4c5d-b810-2030e6b36fb5"
          />
        </span>
        <span
          className="MuiTouchRipple-root-92"
        />
      </span>
      <span
        className="MuiTypography-root-48 MuiTypography-body1-57 MuiFormControlLabel-label-19"
      >
        Virtual
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root-14"
    >
      <span
        className="MuiButtonBase-root-36 MuiIconButton-root-30 MuiPrivateSwitchBase-root-26 MuiCheckbox-root-20 MuiCheckbox-colorSecondary-25 MuiPrivateSwitchBase-checked-27 MuiCheckbox-checked-21"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-35"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-39"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <input
            checked={true}
            className="MuiPrivateSwitchBase-input-29"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="b4d2d5e5-7ece-4dbd-b375-67389b7c4f84"
          />
        </span>
        <span
          className="MuiTouchRipple-root-92"
        />
      </span>
      <span
        className="MuiTypography-root-48 MuiTypography-body1-57 MuiFormControlLabel-label-19"
      >
        Override
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root-14"
    >
      <span
        className="MuiButtonBase-root-36 MuiIconButton-root-30 MuiPrivateSwitchBase-root-26 MuiCheckbox-root-20 MuiCheckbox-colorSecondary-25"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-35"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-39"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <input
            checked={false}
            className="MuiPrivateSwitchBase-input-29"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="72010226-6d1f-4354-b57d-15953292dbd6"
          />
        </span>
        <span
          className="MuiTouchRipple-root-92"
        />
      </span>
      <span
        className="MuiTypography-root-48 MuiTypography-body1-57 MuiFormControlLabel-label-19"
      >
        Sealed
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root-14"
    >
      <span
        className="MuiButtonBase-root-36 MuiIconButton-root-30 MuiPrivateSwitchBase-root-26 MuiCheckbox-root-20 MuiCheckbox-colorSecondary-25"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-35"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-39"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <input
            checked={false}
            className="MuiPrivateSwitchBase-input-29"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="72010226-6d1f-4354-b57d-15953292dbd6"
          />
        </span>
        <span
          className="MuiTouchRipple-root-92"
        />
      </span>
      <span
        className="MuiTypography-root-48 MuiTypography-body1-57 MuiFormControlLabel-label-19"
      >
        I don't know
      </span>
    </label>
  </div>
  <p
    className="MuiFormHelperText-root-84"
  >
    Be careful
  </p>
</div>
`;
